### ISSUE TITLE
[Enhancement] Toggle Ligatures ON/OFF

### ISSUE THREAD

**Issue by Firestar-Reimu** [spyder-ide/spyder#22470]
I am a man who do not want ligatures (so that what I type is what I see)

for example, in python `__init__` with ligatures can be confused with `_init_`

So I am glad to have a checkbox to toggle ligatures ON/OFF

It is better to have maybe a complete setting of font ligatures and stylistic sets, like in vscode, have a long box to type `"editor.fontLigatures": "'ss01', 'ss02', 'ss03', 'ss19', 'calt'"`

**Comment by ccordoba12** [comment]
Hey @Firestar-Reimu, thanks for reporting. I don't know if this is possible in Qt. It seems like it but only in Qt6:

https://doc.qt.io/qt-6/qfont.html#setFeature

However, we don't have time to take a look at this kind of feature because it's too specialized.

**Comment by Firestar-Reimu** [comment]
Does spyder 6 use qt6?

**Comment by ccordoba12** [comment]
No, it still uses Qt5. We'll try to add support for Qt6 later this year or the next one.

**Comment by reachout-sreeram** [comment]
Hi @ccordoba12,

Since Spyder now supports Qt6, I can implement this using the `QFont::setFeature` API (introduced in Qt 6.7). 

I'd like to add an **"Enable font ligatures in the editor"** checkbox under **Preferences > Appearance > Fonts**. In older Qt/Qt5 environments, the checkbox will simply be disabled.

Please let me know about this approach?
